python_tests/pytest_input_generators.py

```diff
@@ -289,6 +345,169 @@ def define_tensor_error_generator(
         yield SampleInput(input_tensor, **es.kwargs), es.ex_type, es.ex_str
 
 
+def take_along_axis_generator(
+    op: OpInfo, dtype: torch.dtype, requires_grad: bool = False, **kwargs
+):
+    make_arg = partial(
+        make_tensor, device="cuda", dtype=dtype, requires_grad=requires_grad
+    )
+    make_index = partial(
+        make_tensor, device="cuda", dtype=torch.long, requires_grad=False
+    )
+
+    # a.shape, dim, b.shape
+    cases = (
+        ((4, 2, 3), 0, (8, 2, 3)),
+        ((4, 2, 3), 1, (4, 1, 3)),
+        ((4, 2, 3), 2, (4, 2, 5)),
+        ((4,), 0, (8)),
+        ((4,), 0, (1)),
+        ((4, 1), 0, (3, 1)),
+        ((4, 1), 1, (4, 5)),
+    )
+
+    for shape_a, dim, shape_b in cases:
+        a = make_arg(shape_a)
+        b = make_index(shape_b, low=0, high=shape_a[dim])
+        yield SampleInput(a, b, dim)
+
+
+def take_along_axis_error_generator(
+    op: OpInfo, dtype: torch.dtype, requires_grad: bool = False, **kwargs
+):
+    # torch.gather(input: Tensor, dim: int, index: LongTensor)
+    # * input and index tensors have same ndims.
+    # * index tensors must be smaller than input tensor along all dims except specified axis.
```

Comment:
We have a slightly different protocol here. We stick with numpy.take_along_axis and expect the shapes to be `broadcast` compatible across all dims, except the given axis